src/views/Header.js

```diff
@@ -99,11 +80,57 @@ class Header extends React.Component<DefaultProps, HeaderProps, *> {
     );
   }
 
+  _getHeaderTitle(navigation: Navigation): ?string {
+    const header = this.props.router.getScreenConfig(navigation, 'header');
+    if (header && header.title) {
+      return header.title;
+    }
+    return this.props.router.getScreenConfig(navigation, 'title');
+  }
+
+  _getHeaderTintColor(navigation: Navigation): ?string {
+    const header = this.props.router.getScreenConfig(navigation, 'header');
+    if (header && header.tintColor) {
+      return header.tintColor;
+    }
+    return undefined;
+  }
+
+  _renderTitleComponent = (props: SubViewProps) => {
+    const color = this._getHeaderTintColor(props.navigation);
+    const title = this._getHeaderTitle(props.navigation);
+    return <HeaderTitle style={color && ({ color })}>{title}</HeaderTitle>;
+  };
+
+  _renderLeftComponent = (props: SubViewProps) => {
+    if (props.scene.index === 0 || !props.onNavigateBack) {
+      return null;
+    }
+    const tintColor = this._getHeaderTintColor(props.navigation);
```

Comment:
Since the `tintColor` is passed by header, while the card stack seems to get `navigationOptions.header.left` and `navigationOptions.header.right` how would we pass `tintColor` to them?